process/transactionLog/failedTxLogsAccumulator.go

```diff
@@ -0,0 +1,109 @@
+package transactionLog
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-go/process"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type logData struct {
+	tx   data.TransactionHandler
+	logs []*vmcommon.LogEntry
+}
+
+type failedTxLogsAccumulator struct {
+	mut     sync.RWMutex
+	logsMap map[string]*logData
+}
+
+// NewFailedTxLogsAccumulator returns a new instance of failedTxLogsAccumulator
+func NewFailedTxLogsAccumulator() *failedTxLogsAccumulator {
+	return &failedTxLogsAccumulator{
+		logsMap: make(map[string]*logData),
+	}
+}
+
+// GetLogs returns the accumulated logs for the provided txHash
+func (accumulator *failedTxLogsAccumulator) GetLogs(txHash []byte) (data.TransactionHandler, []*vmcommon.LogEntry, bool) {
+	if len(txHash) == 0 {
+		return nil, nil, false
+	}
+
+	logsData, found := accumulator.getLogDataCopy(txHash)
+
+	if !found {
+		return nil, nil, found
+	}
+
+	return logsData.tx, logsData.logs, found
+}
+
+func (accumulator *failedTxLogsAccumulator) getLogDataCopy(txHash []byte) (logData, bool) {
+	accumulator.mut.RLock()
+	defer accumulator.mut.RUnlock()
+
+	logsData, found := accumulator.logsMap[string(txHash)]
+	if !found {
+		return logData{}, found
+	}
+
+	logsDataCopy := logData{
+		tx: logsData.tx,
+	}
+
+	logsDataCopy.logs = append(logsDataCopy.logs, logsData.logs...)
+
+	return logsDataCopy, found
+}
+
+// SaveLogs saves the logs into the internal map
+func (accumulator *failedTxLogsAccumulator) SaveLogs(txHash []byte, tx data.TransactionHandler, logs []*vmcommon.LogEntry) error {
```

Comment:
since we are not saving the logs to persistent storage, is it possible that we lose unintended failed tx logs in case of a restart? or the logs will be saved later via tx logs processor if needed?

Reply:
I don’t think we can lose logs. If something fails during tx execution and we somehow lose this, the tx won’t be seen as executed too.